Firestore/core/src/remote/serializer.cc

```diff
@@ -1451,29 +1451,21 @@ std::unique_ptr<WatchChange> Serializer::DecodeExistenceFilterWatchChange(
 
 ExistenceFilter Serializer::DecodeExistenceFilter(
     const google_firestore_v1_ExistenceFilter& filter) const {
-  if (filter.has_unchanged_names) {
-    pb_bytes_array_t* bitmap_ptr = filter.unchanged_names.has_bits
-                                       ? filter.unchanged_names.bits.bitmap
-                                       : nullptr;
-    std::vector<uint8_t> bitmap(bitmap_ptr->bytes,
-                                bitmap_ptr->bytes + bitmap_ptr->size);
-
-    int32_t padding = filter.unchanged_names.has_bits
-                          ? filter.unchanged_names.bits.padding
-                          : 0;
-    int32_t hash_count = filter.unchanged_names.hash_count;
-
-    StatusOr<BloomFilter> maybe_bloom_filter =
-        BloomFilter::Create(bitmap, padding, hash_count);
-
-    if (maybe_bloom_filter.ok()) {
-      return {filter.count, std::move(maybe_bloom_filter).ValueOrDie()};
-    } else {
-      LOG_WARN("Creating BloomFilter failed: %s",
-               maybe_bloom_filter.status().error_message());
-    }
+  if (!filter.has_unchanged_names) {
+    return {filter.count, absl::nullopt};
+  }
+
+  int32_t hash_count = filter.unchanged_names.hash_count;
+  int32_t padding = 0;
+  std::vector<uint8_t> bitmap;
+  if (filter.unchanged_names.has_bits) {
+    padding = filter.unchanged_names.bits.padding;
+
+    pb_bytes_array_t* bitmap_ptr = filter.unchanged_names.bits.bitmap;
+    bitmap = std::vector<uint8_t>(bitmap_ptr->bytes,
```

Comment:
Verify that `bitmap_ptr` and `bitmap_ptr->bytes` are not `nullptr` before accessing them. Otherwise, if they are null, then the program will crash.

Reply:
c++ suggested `bitmap_ptr->bytes !=nullptr` is always true.